toqito/matrices/gen_gell_mann.py

```diff
@@ -1,10 +1,12 @@
 """Generalized Gell-Mann matrices."""
 
+
 import numpy as np
-from scipy import sparse
 
 
-def gen_gell_mann(ind_1: int, ind_2: int, dim: int, is_sparse: bool = False) -> np.ndarray | sparse.lil_matrix:
```

Comment:
@Bchass Thank you for this PR! Is it fine if we merge this PR next week?

I want to understand the difference between using `scipy.sparse` vs. `np.ndarray`. I am a little bit confused about how the return type was previously `matrix` when the `scipy.sparse` [documentation](https://docs.scipy.org/doc/scipy/reference/sparse.html) says they have moved to using arrays. 

Considering other functions in `toqito` also use `scipy.sparse`, I might have to create additional issues for those that require similar changes. For example, see below:

https://github.com/vprusso/toqito/blob/2714b92648a84b5b8b09bb35aab7f31da9fbc975/toqito/perms/symmetric_projection.py#L12

Reply:
Merging next week is totally fine. My initial thought after this refactor is what you already hit on - need to remove all sparse matrix classes instead of converting a matrix to an array. I think it would be a better approach to refactoring all to use an array from the start.

Reply:
Sounds like a plan to me, @Bchass . Do let us know if we can provide any additional input or help, and thank you again!